From 36d93a8e187b7cfd85d8e93bb8c9a3750c5cedb2 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Thu, 19 May 2022 19:36:11 +0300
Subject: [PATCH] [SW-83569] rom: use new boot info fields

In order to improve future compatibility, additional fields have
been added as a part of boot info header structure. This patch
implements the usage of these fields in arcboot rom.

1) ptrs_sz: size of boot info header structure
2) offsets_arr_off: the offsets array offset from the base of ptrs

Change-Id: I6579ba395220bc727233e11da832d3cb62b13a15
---

diff --git a/common/boot_info.c b/common/boot_info.c
index f46bb1a..a9713a5 100644
--- a/common/boot_info.c
+++ b/common/boot_info.c
@@ -79,6 +79,8 @@
 	boot_info.ptrs.magic = HL_BOOT_INFO_MAGIC;
 	boot_info.ptrs.desc_ver = hl_get_boot_info_ver();
 	boot_info.ptrs.desc_sz = sizeof(struct hl_boot_info_desc);
+	boot_info.ptrs.ptrs_sz = sizeof(struct hl_boot_info_ptrs);
+	boot_info.ptrs.offsets_arr_off = boot_info.ptrs.ptrs_sz;
 
 	/* Data offset */
 	offset = sizeof(struct hl_boot_info_ptrs) +
@@ -187,7 +189,7 @@
 	if (ptrs.magic != HL_BOOT_INFO_MAGIC)
 		return 1;
 
-	offsets = (uint32_t *)offset2ptr(sizeof(struct hl_boot_info_ptrs),
+	offsets = (uint32_t *)offset2ptr(ptrs.offsets_arr_off,
 				    BOOT_INFO_BASE);
 
 	for (i = 0; i < ptrs.count; i++) {
